### ISSUE TITLE
Blur effect needed on transparency

### ISSUE THREAD

**Issue by mfreeman72** [linuxmint/cinnamon#9174]
```
 * Cinnamon version 4.4.8
 * Distribution - Mint 19.3
 * Graphics hardware - AMD A8 GPU (Radeon R5) with current open source driver
 * 64 bit
 ```

**Issue**
Any time a transparency is used (especially panels and panel menus), any text or hard lines/textures behind it make text on the menu difficult to read. I would like to request an optional blurred effect behind transparencies. I know that this has been discussed in the past, and it used to be possible to put a bump map image behind it (that's not necessarily what I'm looking for, but it had a similar result), but that issue has been closed with no resolution. I would like to see just a simple, basic blur implemented. Other desktops have this feature, and it not only looks really nice, but makes text on transparent menus and panels much more readable by softening hard lines behind them. Please consider this for future Cinnamon versions.

**Comment by brownsr** [comment]
There's a PR in for this.  Still work in progress as there was a stability fault found at the last stage. 

Regards

Simon

**Comment by viliampucik** [comment]
@brownsr Please let me ask you what is the status of the PR? BTW is there a way one can help with the code or its testing?

**Comment by brownsr** [comment]
@viliampucik I have not found time to work on the PR for a while.  You will see it in the list of open PRs, marked WIP.  There were two problems, the first is a segfault.  I never had that, my suspicion is that it might be caused by having two simultaneous such effects (blur and bumpmap) on at the same time.   A case I never tried during development. If so then that would not probably be that hard to debug.The other was that as a screen snapshot effect, it was very 'busy' on something like the menu where it reacted to menu bar highlighting and pointer movement.  I had planned to research the menu to find if there was a more appropriate piece of CSS to try it on, as ideally it should be applied just to the menu background and not to the active menu elements such as the current menu item or the mouse.  If you want to pick up on this and send any resulting changes to me then that might be very helpful.

Regards

Simon

**Comment by viliampucik** [comment]
Many thanks @brownsr for the details, really appreciated! I am going to familiarize myself with the internals and hopefully test and submit PR in the near future.

Recently I have switched from Deepin to Cinnamon and I really like the work that has been done in Cinnamon. The transparent and bluer effect offered by Deepin and UKUI make those environments stood out. If Cinnamon provides such eye candy as well, it may attract more users and contributors to this nice project.

**Comment by nelwa** [comment]
Are there any new developments regarding this feature?  I would also like this eye candy blur effect to be available.

**Comment by R3tr0BoiDX** [comment]
Me tooooo

**Comment by TPO100Cofficial** [comment]
> Many thanks @ brownsr for the details, really appreciated! I am going to familiarize myself with the internals and hopefully test and submit PR in the near future.

hey, @viliampucik , any progress on that? not to be annoying, if not that's fine

**Comment by Janne-d-Arc** [comment]
I would really like this feature, I think it greatly improves the look of transparency. I like it so much that as a "cheap" workaround I use the "Transparent panels" extension and applied a blur on the area covered by the panel directly on the wallpaper using Gimp obtaining results like these:

![bitmap](https://github.com/linuxmint/cinnamon/assets/13596575/2eea5fd2-af4f-49f3-bb0f-ae9ae09d2a99)

I would love it if one day Cinnamon could make an effect like that without having to edit my wallpapers.

**Comment by Secret-chest** [comment]
It should be optional and decided on by the theme. This is worked on.

**Comment by fredcw** [comment]
@Janne-d-Arc I tried adding `background-blur: 16px;` to the `.panel-bottom` selector (`#panel` selector also works) in the `cinnamon.css` file of the current cinnamon theme while also setting the `background-color` to fully transparent in the same selector. Unfortunately there's a couple of bugs: using it on a side panel (left or right) crashes cinnamon so that's a non starter, and on the top or bottom panel, it works mostly fine except that moving a window next to the panel can sometimes leave some visual artefacts, more noticeable on the bottom panel than the top.

![Screenshot from 2023-09-04 06-18-12](https://github.com/linuxmint/cinnamon/assets/58893963/0da1fdfa-745e-412a-91c5-a6300e1941f5)

**Comment by Janne-d-Arc** [comment]
@fredcw I knew that since at least 2012 tests were being done with transparencies ([https://forums.linuxmint.com/viewtopic.php?t=97101](url)) but I didn't know that this was already implemented at in an experimental way! Too bad it's still unstable but I'll still do some tests with the settings you mention, thank you very much!

**Comment by mfreeman72** [comment]
@fredcw Awesome! I'm glad this is being worked on! The results are actually pretty beautiful, as long as it's not in one of the crash conditions you mentioned, and it's not being interacted with. Hopefully it will become stable enough for everyday use sometime soon.

**Comment by JosephMcc** [comment]
As far as I know, no one is working on this. This is still the same PR that was merged quite a long time ago and frankly, is broken and nowhere being in any sort of usable state.

**Comment by mfreeman72** [comment]
That's unfortunate. Blur makes transparency actually usable and readable. Without it, any text or hard-line objects behind the transparency make text on top of the transparency difficult to read, if not completely unreadable. Blur is, in my opinion, an absolute must-have on transparent themes. Without it, transparency hinders, if not utterly destroys usability.

**Comment by JosephMcc** [comment]
@mfreeman72 I agree. Transparency is pretty much useless without it. I will add, and it's only my opinion, that I find the whole blurred transparency things a bit of a gimmick. I wouldn't be interested in it even if it worked except in very small doses.

**Comment by mfreeman72** [comment]
@JosephMcc To each his own, I guess. I personally love the look of a blurred transparent background, if done right. But I'm more of an artist/musician than a technical user, so I prefer things to be as visually and aurally appealing as possible, more than technically effective.

**Comment by sakralbar** [comment]
It seems to me that the blur effect is what cinnamon lacks in order to look a little more cute and modern.

**Comment by mfreeman72** [comment]
@fredcw 
>  it works mostly fine except that moving a window next to the panel can sometimes leave some visual artefacts, more noticeable on the bottom panel than the top.

Looking at how some have implemented this in Gnome, they also had artifacts. They got around it by creating a .drirc file in the user's home folder, and putting the following in it:

`<device screen="0" driver="dri2">
    <application name="Default">
        <option name="vblank_mode" value="0"/>
    </application>
</device>`

This doesn't work for NVIDIA GPU's, but they had a different solution for that.

Anyway, I tried this in Cinnamon, and it pretty much solves the artifacting problems, except that a small line appears on either side of panel applets when the mouse moves over them. But they go away quickly. So it's pretty close to being a non-issue. Now I just need to experiment to see if this can also be implemented in panel menus...

**Comment by mfreeman72** [comment]
Yep, it does work, and it looks really nice! However, the same line surrounds things the mouse moves over in some menus, and it's definitely more pronounced. If they could fix this minor issue, then it would be perfectly usable.

**Comment by OfficialPixelBrush** [comment]
Any news on this end?

**Comment by mfreeman72** [comment]
As far as I know, per comments above, this is not being worked on, so it's in limbo until the devs decide it's worth looking at. It would be awesome if I had better news, but sadly I don't.

**Comment by echuber2** [comment]
Is there still an active PR with some work done? I thought I had looked at one that did an iterative box blur and I thought, it might be more performant to do a separable gaussian blur.

**Comment by ironhak** [comment]
Cinnamon clearly needs some sort of revamp because it's looking really old compared to gnome or KDE. The fun thing is that it could be as simple as:
- implement a proper blurred panels effect, without artifacts as mentioned here. 
- make a new theme with blurred panels (and optional: blurred window title bar)
- https://cinnamon-spices.linuxmint.com/extensions/view/72 insert this into cinnamon by default

**Comment by mfreeman72** [comment]
Check this out: https://github.com/klangman/BlurCinnamon

It only blurs the panel (but not panel menus), expo, and overview, and if your panel is on the bottom, it doesn't see windows that are behind the panel (that's being worked on), but otherwise it works well and is faster and less glitchy than anything posted here so far. A combination of this extension and other solutions posted here might produce the effects we're hoping for in this GitHub issue

**Comment by fre3e3e** [comment]
> Cinnamon clearly needs some sort of revamp because it's looking really old compared to gnome or KDE. The fun thing is that it could be as simple as:
> 
>     * implement a proper blurred panels effect, without artifacts as mentioned here.
> 
>     * make a new theme with blurred panels (and optional: blurred window title bar)
> 
>     * https://cinnamon-spices.linuxmint.com/extensions/view/72 insert this into cinnamon by default

100% agree. The biggest issue are the artifacts when hovering menu elements if those could be fixed, would make it useable instantly.

**Comment by vyvir** [comment]
I agree that blur should be added. Blur in GNOME looks gorgeous, and it would, in my opinion, make Cinnamon look so much better and more modern.

The theme below is a showcase for such a feature. In Cinnamon, it currently makes the windows only transparent, but it would look gorgeous if window blur was added as an option to Cinnamon.

https://github.com/owozsh/myopia

**Comment by MythicHH** [comment]
Blur would be awesome.

**Comment by Akopov4** [comment]
Completely agree. Please add blur effect.
> That's unfortunate. Blur makes transparency actually usable and readable. Without it, any text or hard-line objects behind the transparency make text on top of the transparency difficult to read, if not completely unreadable. Blur is, in my opinion, an absolute must-have on transparent themes. Without it, transparency hinders, if not utterly destroys usability.

**Comment by KrystianMazurkiewicz** [comment]
I agree with this suggestion. Blur/transparency support would make Cinnamon feel more polished and would help people who want a more modern desktop style.

Here is an example of the kind of look this could make possible on Mint ([Mystical-Blue-Theme](https://github.com/juxtopposed/Mystical-Blue-Theme)):

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4f031dfe-f324-4f42-a127-43b42bad18b0" />

**Comment by JosephMcc** [comment]
This is not longer the place for feature requests and this has been brought up plenty. Going forward all requests go here: https://github.com/orgs/linuxmint/discussions
